go/test/endtoend/cluster/mysqlctld_process.go

```diff
@@ -168,3 +168,8 @@ func (mysqlctld *MysqlctldProcess) IsHealthy() bool {
 	_, err := mysql.Connect(context.Background(), &params)
 	return err == nil
 }
+
+// HasShutdown checks if the process has been set to nil
+func (mysqlctld *MysqlctldProcess) HasShutdown() bool {
+	return mysqlctld.process == nil
```

Comment:
I don't see where setting that to nil happens. Am I missing it?

Reply:
> I don't see where setting that to nil happens. Am I missing it?

In a goroutine started in  `MysqlctldProcess.Start()` which waits for it to exit and then sets it to nil. That's where the flakiness is: when it tries to restart it, it fails the nil check.